owncloudApp/src/main/java/com/owncloud/android/presentation/ui/sharing/fragments/ShareFileComposeFragment.kt

```diff
@@ -0,0 +1,626 @@
+                                }
+                            )
+                        }
+                        if (privateShares.isNullOrEmpty()) {
+                            item { EmptyListText(text = stringResource(id = R.string.share_no_users)) }
+                        } else {
+                            privateShares = privateShares!!.sortedBy { it.sharedWithDisplayName }
+                            items(privateShares!!) { share ->
+                                ShareUserItem(
+                                    share = share,
+                                    unshare = {
+                                        // Unshare
+                                        Timber.d("Removing private share with ${share.sharedWithDisplayName}")
+                                        listener?.showRemoveShare(share)
+                                    },
+                                    edit = {
+                                        // Move to fragment to edit share
+                                        Timber.d("Editing ${share.sharedWithDisplayName}")
+                                        listener?.showEditPrivateShare(share)
+                                    }
+                                )
+                                Divider(color = colorResource(id = R.color.list_divider_background))
+                            }
+                        }
+                    }
+
+                    // Hide share via link section if it is not enabled or if share API or public share are not enabled
+                    if (shareViaLinkAllowed && isShareApiEnabled && isPublicShareEnabled) {
+                        item {
+                            // Show or hide button for adding a new public share depending on the capabilities and the server version
+                            SectionHeader(
+                                title = stringResource(id = R.string.share_via_link_section_title),
+                                showAddButton = isMultiplePublicSharingEnabled || (!isMultiplePublicSharingEnabled && publicLinks.isNullOrEmpty()),
+                                onClickAddButton = {
+                                    // Show Add Public Link Fragment
+                                    listener?.showAddPublicShare(availableDefaultPublicName)
+                                }
+                            )
+                        }
+                        // Hide warning about public links if not enabled
+                        if (shareWarningAllowed) {
+                            item { WarningText() }
+                        }
+                        if (publicShares.isNullOrEmpty()) {
+                            item { EmptyListText(text = stringResource(id = R.string.share_no_public_links)) }
+                        } else {
+                            publicShares = publicShares!!.sortedBy { it.name }
+                            items(publicShares!!) { share ->
```

Comment:
**SPOILER**: If u followed the same path as the `privateShares`, those `!!` 🙀  should not be there

Reply:
*Screams in Kotlin 😱